RiotShareExtension/Modules/Share/Listing/RoomsListViewController.m

```diff
@@ -139,53 +139,34 @@ - (void)showShareAlertForRoomPath:(NSIndexPath *)indexPath
         
-        // The selected room is instanciated here.
-        [[ShareExtensionManager sharedManager].fileStore asyncAccountDataOfRoom:recentCellData.roomSummary.roomId success:^(MXRoomAccountData * _Nonnull accountData) {
-            
-            [[ShareExtensionManager sharedManager].fileStore stateOfRoom:recentCellData.roomSummary.roomId success:^(NSArray<MXEvent *> * _Nonnull roomStateEvents) {
-                
-                MXSession *session = [[MXSession alloc] initWithMatrixRestClient:[[MXRestClient alloc] initWithCredentials:[ShareExtensionManager sharedManager].userAccount.mxCredentials andOnUnrecognizedCertificateBlock:nil]];
-                
-                // To handle correctly the crypto, we have to set a store (use a fake store)
-                __weak MXSession *weakSession = session;
-                [session setStore:[[MXNoStore alloc] init] success:^{
-                    
-                    if (weakSession)
-                    {
-                        __strong MXSession *session = weakSession;
-                        
-                        MXRoom *selectedRoom = [[MXRoom alloc] initWithRoomId:recentCellData.roomSummary.roomId andMatrixSession:session andStateEvents:roomStateEvents andAccountData:accountData];
-                        
-                        [ShareExtensionManager sharedManager].delegate = self;
-                        
-                        [[ShareExtensionManager sharedManager] sendContentToRoom:selectedRoom failureBlock:^(NSError* error) {
-                            
-                            NSString *title;
-                            if ([error.domain isEqualToString:MXEncryptingErrorDomain])
-                            {
-                                title = NSLocalizedStringFromTable(@"share_extension_failed_to_encrypt", @"Vector", nil);
-                            }
-                            
-                            [self showFailureAlert:title];
-                        }];
-                    }
-                    
-                } failure:^(NSError *error) {
-                    
-                    NSLog(@"[RoomsListViewController] failed to prepare matrix session]");
-                    
-                }];
-                
-            } failure:^(NSError * _Nonnull error) {
-                
-                NSLog(@"[RoomsListViewController] failed to get state events");
-                
+        // The selected room is instanciated here
+        MXSession *session = [[MXSession alloc] initWithMatrixRestClient:[[MXRestClient alloc] initWithCredentials:[ShareExtensionManager sharedManager].userAccount.mxCredentials andOnUnrecognizedCertificateBlock:nil]];
+
+        [MXFileStore setPreloadOptions:0];
+
+        MXWeakify(session);
+        [session setStore:[[MXNoStore alloc] init] success:^{
```

Comment:
The session store must be `[ShareExtensionManager sharedManager].fileStore`

Reply:
Thank you!
That explains a bug I had.